// @adyen/api-library/lib/src/typings/platformsAccount/createAccountRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateAccountRequest = void 0;
class CreateAccountRequest {
    static getAttributeTypeMap() {
        return CreateAccountRequest.attributeTypeMap;
    }
}
exports.CreateAccountRequest = CreateAccountRequest;
CreateAccountRequest.discriminator = undefined;
CreateAccountRequest.attributeTypeMap = [
    {
        "name": "accountHolderCode",
        "baseName": "accountHolderCode",
        "type": "string"
    },
    {
        "name": "bankAccountUUID",
        "baseName": "bankAccountUUID",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "payoutMethodCode",
        "baseName": "payoutMethodCode",
        "type": "string"
    },
    {
        "name": "payoutSchedule",
        "baseName": "payoutSchedule",
        "type": "CreateAccountRequest.PayoutScheduleEnum"
    },
    {
        "name": "payoutScheduleReason",
        "baseName": "payoutScheduleReason",
        "type": "string"
    },
    {
        "name": "payoutSpeed",
        "baseName": "payoutSpeed",
        "type": "CreateAccountRequest.PayoutSpeedEnum"
    }
];
(function (CreateAccountRequest) {
    let PayoutScheduleEnum;
    (function (PayoutScheduleEnum) {
        PayoutScheduleEnum["BiweeklyOn1StAnd15ThAtMidnight"] = "BIWEEKLY_ON_1ST_AND_15TH_AT_MIDNIGHT";
        PayoutScheduleEnum["Daily"] = "DAILY";
        PayoutScheduleEnum["DailyAu"] = "DAILY_AU";
        PayoutScheduleEnum["DailyEu"] = "DAILY_EU";
        PayoutScheduleEnum["DailySg"] = "DAILY_SG";
        PayoutScheduleEnum["DailyUs"] = "DAILY_US";
        PayoutScheduleEnum["Hold"] = "HOLD";
        PayoutScheduleEnum["Monthly"] = "MONTHLY";
        PayoutScheduleEnum["Weekly"] = "WEEKLY";
        PayoutScheduleEnum["WeeklyOnTueFriMidnight"] = "WEEKLY_ON_TUE_FRI_MIDNIGHT";
    })(PayoutScheduleEnum = CreateAccountRequest.PayoutScheduleEnum || (CreateAccountRequest.PayoutScheduleEnum = {}));
    let PayoutSpeedEnum;
    (function (PayoutSpeedEnum) {
        PayoutSpeedEnum["Instant"] = "INSTANT";
        PayoutSpeedEnum["SameDay"] = "SAME_DAY";
        PayoutSpeedEnum["Standard"] = "STANDARD";
    })(PayoutSpeedEnum = CreateAccountRequest.PayoutSpeedEnum || (CreateAccountRequest.PayoutSpeedEnum = {}));
})(CreateAccountRequest = exports.CreateAccountRequest || (exports.CreateAccountRequest = {}));
